src/rgw/rgw_rest_s3.cc

```diff
@@ -4177,15 +4177,16 @@ rgw::auth::s3::LocalEngine::authenticate(
 
   const VersionAbstractor::server_signature_t server_signature = \
     signature_factory(cct, k.key, string_to_sign);
+  auto compare = signature.compare(server_signature);
 
   ldout(cct, 15) << "string_to_sign="
                  << rgw::crypt_sanitize::log_content{string_to_sign}
                  << dendl;
   ldout(cct, 15) << "server signature=" << server_signature << dendl;
   ldout(cct, 15) << "client signature=" << signature << dendl;
-  ldout(cct, 15) << "compare=" << signature.compare(server_signature) << dendl;
+  ldout(cct, 15) << "compare=" << compare << dendl;
 
-  if (static_cast<boost::string_view>(server_signature) != signature) {
+  if (compare != 0) {
```

Comment:
@rzarzynski can you take a look at this one please? was there any material difference between `compare()` and `operator!=()`?

Reply:
`VersionAbstractor::server_signature_t` is an alias over our `basic_sstring` template, so it has the overload for conversion to `boost::string_view`. In turn, the `operator!=` of `boost::string_view` internally uses `compare()` enriched with the optimization/early exit for different string sizes.
 
Also, `a != b` appears more readable to me than the `compare()`-based  check.